bedrock/mozorg/templates/mozorg/home/home-new.html

```diff
@@ -46,18 +47,42 @@
   {% endif %}
 {% endblock %}
 
+{% block site_header %}
+  {% if show_fundraising_nav_cta and variation == '1' %}
+    {% with
+      custom_nav_cta="<a class='mzp-c-button mzp-t-secondary c-fx-users' href='https://foundation.mozilla.org/?form=cta-donate-nav-fx'>
+      <span class='mzp-c-button-icon-start'>
+      <svg width='16' height='16' viewBox='0 0 16 16'>
+      <path fill='#ff0000' d='M1.2,2.2c-1.5,1.5-1.5,4.2,0,5.7L8,15c2.3-2.4,4.5-4.7,6.9-7c1.5-1.5,1.5-4.2,0-5.7c-1.5-1.5-4-1.5-5.6,0L8,3.5 L6.7,2.3C5.1,0.6,2.7,0.6,1.2,2.2'/>
+      </svg>
+      </span>Donate</a>
+
+      <a class='mzp-c-button mzp-t-secondary c-non-fx-users' href='https://foundation.mozilla.org/?form=cta-donate-nav'>
+      <span class='mzp-c-button-icon-start'>
+      <svg width='16' height='16' viewBox='0 0 16 16'>
+      <path fill='#ff0000' d='M1.2,2.2c-1.5,1.5-1.5,4.2,0,5.7L8,15c2.3-2.4,4.5-4.7,6.9-7c1.5-1.5,1.5-4.2,0-5.7c-1.5-1.5-4-1.5-5.6,0L8,3.5 L6.7,2.3C5.1,0.6,2.7,0.6,1.2,2.2'/>
+      </svg>
+      </span>Donate</a>" |safe %}
```

Comment:
**suggestion** the indentation here makes the markup a little difficult to read / follow. Might putting this into it's own macro help make it more readable?

```jinja
{% macro donate_button() -%}
<a class='mzp-c-button mzp-t-secondary c-fx-users' href='https://foundation.mozilla.org/?form=cta-donate-nav-fx'>
  <span class='mzp-c-button-icon-start'>
    <svg width='16' height='16' viewBox='0 0 16 16'>
      <path fill='#ff0000' d='M1.2,2.2c-1.5,1.5-1.5,4.2,0,5.7L8,15c2.3-2.4,4.5-4.7,6.9-7c1.5-1.5,1.5-4.2,0-5.7c-1.5-1.5-4-1.5-5.6,0L8,3.5 L6.7,2.3C5.1,0.6,2.7,0.6,1.2,2.2'/>
    </svg>
  </span>
  Donate
</a>

<a class='mzp-c-button mzp-t-secondary c-non-fx-users' href='https://foundation.mozilla.org/?form=cta-donate-nav'>
  <span class='mzp-c-button-icon-start'>
    <svg width='16' height='16' viewBox='0 0 16 16'>
      <path fill='#ff0000' d='M1.2,2.2c-1.5,1.5-1.5,4.2,0,5.7L8,15c2.3-2.4,4.5-4.7,6.9-7c1.5-1.5,1.5-4.2,0-5.7c-1.5-1.5-4-1.5-5.6,0L8,3.5 L6.7,2.3C5.1,0.6,2.7,0.6,1.2,2.2'/>
    </svg>
  </span>
  Donate
</a>
{%- endmacro %}
```

Then you could do:

```jinja
{% with custom_nav_cta=donate_button() %}
  {% include 'includes/protocol/navigation/navigation.html' %}
{% endwith %}
```

Reply:
I thought about your suggestion, and ideally it would make more sense to create the macro with only one donate button rather than the two (one for Fx, one for non-Fx) because not every donate button will have that variation in mind -- this is ideal because macros are created to be usable throughout the codebase. So I went ahead and created a macro for a singular donate button: 

```
{% macro donate_button(id='', url='') -%}
  <a class='mzp-c-button mzp-t-secondary' id="{{ id }}" href='{{ url }}'>
    <span class='mzp-c-button-icon-start'>
      <svg width='16' height='16' viewBox='0 0 16 16'>
        <path fill='#ff0000' d='M1.2,2.2c-1.5,1.5-1.5,4.2,0,5.7L8,15c2.3-2.4,4.5-4.7,6.9-7c1.5-1.5,1.5-4.2,0-5.7c-1.5-1.5-4-1.5-5.6,0L8,3.5 L6.7,2.3C5.1,0.6,2.7,0.6,1.2,2.2'/>
      </svg>
    </span>
    Donate
  </a>
{%- endmacro %}
```

and added it to the HTML file this way:
```
{% with custom_nav_cta=donate_button(id="c-fx-users", url="https://foundation.mozilla.org/?form=cta-donate-nav-fx") %}
      {% include 'includes/protocol/navigation/navigation.html' %}
    {% endwith %}

    {% with custom_nav_cta=donate_button(id="c-non-fx-users", url="https://foundation.mozilla.org/?form=cta-donate-nav") %}
      {% include 'includes/protocol/navigation/navigation.html' %}
    {% endwith %}
```

To note what's happening above, I'm adding 2 nav bars, one holding the donation link for Fx users, and another nav bar holding the link for the non-Fx users. 

However, because I'm trying to show/hide the correct button based on a user's browser, I'm unable to fully hide the nav bar of each variation conditionally, as I'm only able to tackle the button, so I'm getting the following (2 nav bars, one without the donation button instead of just one nav bar with the correct donation button):
![Screenshot 2024-07-12 at 4 23 27 PM](https://github.com/user-attachments/assets/b45601ba-a010-43d8-b501-2a48e534422d)

I could fix this up in Javascript by targeting the parent nodes, but to be honest I think that would be overkill, given this donation CTA is only going to be up for one week and the code will be removed eventually. 

So I'm just going to go ahead and keep the code as-is, but to keep `home-new.html` cleaner, I moved this `custom_nav_cta` macro to a file within the includes folder `home/includes/mofo-fundraising-nav-cta.html`

Reply:
I think this is over-complicating what needs doing. Macros don’t need to live in `macros.html` and be shared amongst the whole code base. It can live directly in your `home-new.html` and copy-pasted directly above the code in question, and only needs to exist for the duration of the experiment. Does that helps to simplify what I suggested?

Reply:
What you have still works and is definitely more readable than before. But it’s still a bit more complicated than what I was trying to suggest 🙂

Reply:
Ah yes okay, that definitely simplified things :) I just automatically assumed you meant `macros.html` 😅